File: GestionRessourcesInfo/src/main/java/com/example/GestionRessourcesInfo/model/Panne.java
package com.example.GestionRessourcesInfo.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Panne {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String description;
    private Date datePanne;
    private boolean resolu;


    @ManyToOne
    @JoinColumn(name = "equipement_id")
    @JsonIgnore
    private Equipement equipement;

    @OneToMany(mappedBy = "panne")
    @JsonIgnore
    private List<TicketDeSupport> tickets;
}

File: FrontEnd/front-endSi/src/app/models/panne.ts
import {Equipement} from "./equipement";
import {TicketDeSupport} from "./ticket-de-support";

export interface Panne {
  id?: number;
  description: string;
  datePanne: string;
  resolu: boolean;
  equipement: { idEquipement: number };
  tickets?: { idTicket: number }[];
}

